### DISCUSSION TITLE
Reload sortspec command

### DISCUSSION THREAD

**Discussion by replete** [SebastianMC/obsidian-custom-sort#63] [Q&A] [answered]
Not really a big issue, but while I'm here I'll mention it:

While I was designing my vault and making sort adjustments, I had to press a lot of keys to reload my sortspec. I improved this by making a `Commander` macro with `sort off` and `sort on`.

It would be a mild improvement to have a single command to turn it off and on again in this scenario.

Auto-reload watching sortspec came to mind also, but Obsidian seems to save/update notes quite frequently and I could imagine this might be get the plugin's knickers in a twist...

**Comment by SebastianMC** [comment, marked as answer]
The plugin supports two commands, which can be invoked from the Obsidian commands palette.
While the full names of the commands are long, they can be invoked by typing in **sort on** or **sort off** (or **sort-on** or **sort-off**) thanks to the convenience of Obsidian UI:

![custom sort commands](https://user-images.githubusercontent.com/23032356/221904742-1c68c2d3-fcc4-4244-94d3-f690d77eb70d.jpg)

- **sort on**, as the description states, (re)parses the sorting specification and applies it, regardless of current plugin state
- **sort off**, suspends the custom sorting

I will add description of the commands to plugin documentation

**Comment by syrth-io** [comment]
(deleted original comment since I missed the mention of Commander in your post - sorry! very out of it today.)

this is very very late, but I also did the same for my Custom File Sort, and actually also added a "save file" command at the beginning of the macro before the turn off/on custom sort as I usually only refresh after editing the sortspec file for a bit of insurance that the file has actually saved before invoking the sorting commands. it isn't quite the same as an automatic watch, and likely not as relevant now that drag-and-drop sorting has been implemented (though I personally enjoy maintaining a sortspec file, for folders at least), but it's saved me some frustration when being between autosaves.

**Comment by SebastianMC** [comment]
Hi @syrth-io,

taking the opportunity, some technical details under the hood:
- the newly introduced drag and drop via bookmarks integration doesn't exclude nor overwrite any of previous capabilities of the plugin. It seamlessly integrates with what was there before. You can sort some files, some folders, a subset of files, etc. via drag and drop, and at the same time (even within the same folder) use sortspec files
- technically, the integration with bookmarks injects on-the-fly the below sorting specification:
  ```yaml
  ---
  target-folder: /*
  bookmarked:
    < by-bookmarks-order
  sorting: standard
  ---
  ```
  the specification is applied to each folder of the vault (`target-folder: /*`) unless an explicit `sorting-spec:` is used for the folder.   It tells to collect bookmarked items in the top of a folder (`bookmarked:`) in the order reflecting bookmarks (`  < by-bookmarks-order`). Then goes the rest of files and folders (not bookmarked ones) ordered as selected in obsidian UI (`sorting: standard`).
- the custom sort plugin whether suspended or active, remains passive and doesn't perform any activity. In other words it doesn't monitor file changes, doesn't process file edits, etc. So the only reason to issue the command `sort off` is when you want to see the File Explorer in its standard state, standard sorting.
- when the command `sort on` is executed, it triggers reading and parsing of sorting specifications. In other words, the sorting specifications are read and parsed once when the `sort on` command is executed. And only then. Clicking the ribbon toggle icon under the hood executes the logic of `sort on` or `sort off` commands.
- in other words, you can edit, update, even remove the sorting specifications, it will not trigger any action in the plugin, and will have no influence on the order of items displayed in File Explorer, until you explicitly invoke the `sort on` command. This is an intentional performance optimisation (with the obvious drawback of requiring explicit `sort on`). In my personal use, I have a keyboard shortcut for `sort on` and I press it after I edit the sorting specification, to refresh the view of File Explorer
- after `sort on` was executed with success, the plugin remains inactive. It is now the Obsidian File Explorer code responsibility to invoke the sorter module on folder(s), when there is a need, in reaction to events.
